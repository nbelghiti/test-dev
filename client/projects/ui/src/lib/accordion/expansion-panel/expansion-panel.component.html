<div
	class="c-expansion-panel {{ level }}"
	[ngClass]="{
		'is-open': expanded,
		'c-expansion-panel--with-spacing': withSpacing
	}"
>
	<div class="c-expansion-panel__header" (click)="toggleExpansion(!expanded)">
		<span class="c-expansion-panel__title">{{title}}</span>

		<div class="c-expansion-panel__actions">
			<div class="c-card__actions" #actions>
				<ng-content select=[uiExpansionPanelAction]></ng-content>
			</div>

			<span class="c-expansion-panel__icon">
				<ui-icon *ngIf="level === ExpansionPanelLevel.primary; else otherLevel" [name]="IconName.caretDown"></ui-icon>

				<ng-template #otherLevel>
					<ui-icon [name]="expanded ? IconName.minusSquare : IconName.plusSquare" [size]="IconSize.small"></ui-icon>
				</ng-template>
			</span>
		</div>
	</div>

	<div *ngIf="expanded" class="c-expansion-panel__content">
		<ng-content></ng-content>
	</div>
<div>
