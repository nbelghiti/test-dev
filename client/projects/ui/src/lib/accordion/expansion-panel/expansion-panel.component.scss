// =============================================================================
// :: Imports
// =============================================================================
@import "global";

// =============================================================================
// :: Settings
// =============================================================================
$padding-base: 0.6rem 2.5rem 0.6rem 2rem;

// =============================================================================
// :: Expansion panel
// =============================================================================
.c-expansion-panel {
	border: $border-width-base solid $border-color-grey-86;
}

// =============================================================================
// :: Elements
// =============================================================================
.c-expansion-panel__header {
	display: flex;
	justify-content: space-between;
	align-items: center;
	padding: $padding-base;

	background-color: $background-color-grey-86;

	cursor: pointer;
}

.c-expansion-panel__title {
	font-size: $font-size-base;
	font-weight: $font-weight-bold;
	line-height: $line-height-base;

	color: $text-color-base;
}

.c-expansion-panel__icon {
	display: flex;
	align-items: center;
	max-height: 1.6rem;
	margin-left: 1rem;

	transition: $animation-base;
}

.c-expansion-panel__actions {
	display: flex;
	align-items: center;
}

// =============================================================================
// :: Modifiers
// =============================================================================
.c-expansion-panel--secondary,
.c-expansion-panel--tertiary {

	.c-expansion-panel__icon {
		color: $text-color-grey-46;
	}
}

.c-expansion-panel--secondary {

	.c-expansion-panel__header {
		background-color: $background-color-grey-96;
	}
}

.c-expansion-panel--tertiary {

	.c-expansion-panel__header {
		background-color: $background-color-base;
	}
}

.c-expansion-panel--with-spacing {

	.c-expansion-panel__content {
		padding: $padding-base;
	}
}

// =============================================================================
// :: States
// =============================================================================
.c-expansion-panel {

	&.is-open {

		box-shadow: 0 0 0 0.3rem rgba($box-shadow-color-base, 0.1);

		&:not(.c-expansion-panel--secondary):not(.c-expansion-panel--tertiary) {

			.c-expansion-panel__icon {
				transform: rotate(180deg);
			}
		}

		.c-expansion-panel__icon {
			color: $text-color-base;
		}
	}
}
