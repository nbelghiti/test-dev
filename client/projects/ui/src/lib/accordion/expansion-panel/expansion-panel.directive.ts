import { Directive, ElementRef, HostListener } from '@angular/core';

@Directive({
	selector: '[uiExpansionPanelAction]'
})

export class ExpansionPanelActionDirective {
	constructor(
		private el: ElementRef
	) {
		this.el.nativeElement.classList.add('u-margin-left-1x');
	}

	@HostListener('click', ['$event']) onClick(e: MouseEvent) {
		this.handleActions(e);
	}

	private handleActions(e: MouseEvent): void {
		e.preventDefault();
		e.stopPropagation();
	}
}
