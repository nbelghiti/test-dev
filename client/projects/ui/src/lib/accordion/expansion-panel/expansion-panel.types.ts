export enum ExpansionPanelLevel {
	primary = '',
	secondary = 'c-expansion-panel--secondary',
	tertiary = 'c-expansion-panel--tertiary',
}
