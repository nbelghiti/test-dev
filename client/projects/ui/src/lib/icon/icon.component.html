<i class="c-icon__glyph {{ size && 'c-icon__glyph--' + size }} {{ type && 'c-icon__glyph--' + type }} fas fa-{{ name }}"></i>
