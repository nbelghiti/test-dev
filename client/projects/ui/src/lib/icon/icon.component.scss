// =============================================================================
// :: Imports
// =============================================================================
@import "global";

// =============================================================================
// :: Icon
// =============================================================================
:host {
	display: inline-flex;
	justify-content: center;
	align-items: center;

	font-size: inherit;
}

// =============================================================================
// :: Elements
// =============================================================================
.c-icon__glyph {
	line-height: inherit;

	cursor: pointer;
}

// =============================================================================
// :: Modifiers
// =============================================================================
.c-icon__glyph--large {
	font-size: 2rem;
}

.c-icon__glyph--small {
	font-size: 1.4rem;
}

.c-icon__glyph--tiny {
	font-size: 1.2rem;
}

.c-icon__glyph--success {
	color: $ui-color-success-dark;
}

.c-icon__glyph--error {
	color: $ui-color-error-dark;
}

.c-icon__glyph--warning {
	color: $ui-color-warning-dark;
}

.c-icon__glyph--info {
	color: $ui-color-info-dark;
}
