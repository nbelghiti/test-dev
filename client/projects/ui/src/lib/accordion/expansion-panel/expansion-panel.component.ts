import {
	ChangeDetectorRef,
	Component,
	EventEmitter,
	Input,
	Output,
	ViewChild
} from '@angular/core';

import { ButtonType } from '../../button/button.types';
import { IconName, IconSize } from '../../icon/icon.types';

import { ExpansionPanelLevel } from './expansion-panel.types';

@Component({
	selector: 'ui-expansion-panel',
	templateUrl: './expansion-panel.component.html',
	styleUrls: ['./expansion-panel.component.scss']
})
export class ExpansionPanelComponent {
	@ViewChild('actions', { static: false }) actions = null;
	@Input() public expanded = false;
	@Input() public title: string;
	@Input() public level: ExpansionPanelLevel = ExpansionPanelLevel.primary;
	@Input() public withSpacing: boolean = true;
	@Output() public toggle: EventEmitter<boolean> = new EventEmitter<boolean>();

	public ExpansionPanelLevel: typeof ExpansionPanelLevel = ExpansionPanelLevel;
	public IconName: typeof IconName = IconName;
	public IconSize: typeof IconSize = IconSize;
	public ButtonType: typeof ButtonType = ButtonType;
	public hasActions: boolean = false;

	constructor(
		private ref?: ChangeDetectorRef
	) {}

	public toggleExpansion(state: boolean): void {
		this.toggle.emit(state);
	}
}
