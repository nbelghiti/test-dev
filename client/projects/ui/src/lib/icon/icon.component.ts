import { Component, HostBinding, Input } from '@angular/core';

import { IconName, IconSize, IconType } from './icon.types';

@Component({
	selector: 'ui-icon',
	templateUrl: './icon.component.html',
	styleUrls: ['./icon.component.scss']
})

export class IconComponent {
	@Input() public name: IconName;
	@Input() public type: IconType = null;
	@Input() public size: IconSize = null;

	@HostBinding('class.c-icon') public className = true;
}
