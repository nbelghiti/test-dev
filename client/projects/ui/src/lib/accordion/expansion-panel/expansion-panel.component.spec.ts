import { ComponentFixture, TestBed, async } from '@angular/core/testing';

import { IconComponent } from '../../icon/icon.component';

import { ExpansionPanelComponent } from './expansion-panel.component';

describe('ExpansionPanelComponent', () => {
	let component: ExpansionPanelComponent;
	let fixture: ComponentFixture<ExpansionPanelComponent>;

	beforeEach(async(() => {
		TestBed.configureTestingModule({
			declarations: [
				ExpansionPanelComponent,
				IconComponent,
			]
		})
		.compileComponents();
	}));

	beforeEach(() => {
		fixture = TestBed.createComponent(ExpansionPanelComponent);
		component = fixture.componentInstance;
		fixture.detectChanges();
	});

	it('should create', () => {
		expect(component).toBeTruthy();
	});

	it('Should render correctly', () => {
		expect(fixture).toMatchSnapshot();
	});
});
