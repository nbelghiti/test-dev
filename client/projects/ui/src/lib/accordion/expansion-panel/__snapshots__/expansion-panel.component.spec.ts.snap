// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExpansionPanelComponent Should render correctly 1`] = `
<ui-expansion-panel
  ButtonType={[Function Object]}
  ExpansionPanelLevel={[Function Object]}
  IconName={[Function Object]}
  IconSize={[Function Object]}
  actions={[Function ElementRef]}
  expanded="false"
  hasActions="false"
  level=""
  ref={[Function ViewRef_]}
  toggle={[Function EventEmitter]}
  withSpacing={[Function Boolean]}
>
  <div
    class="c-expansion-panel c-expansion-panel--with-spacing"
    ng-reflect-klass="c-expansion-panel "
    ng-reflect-ng-class="[object Object]"
  >
    <div
      class="c-expansion-panel__header"
    >
      <span
        class="c-expansion-panel__title"
      >
        
      </span>
      <div
        class="c-expansion-panel__actions"
      >
        <div
          class="c-card__actions"
        />
        <span
          class="c-expansion-panel__icon"
        >
          
          <ui-icon
            class="c-icon"
            ng-reflect-name="caret-down"
          >
            <i
              class="c-icon__glyph   fas fa-caret-down"
            />
          </ui-icon>
          
        </span>
      </div>
    </div>
    
    <div />
  </div>
</ui-expansion-panel>
`;
